### DISCUSSION TITLE
Open Library: Call Number

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#148] [Ideas]
**congratulations for the release of the new version!**

Note: the query with Open Library, fills the field Call number.
this field is specific to each library and should not be filled against the user's will the standard fields are rather 
LC Control Number and Dewey Class Number.

suggestion: add a Call Number option via parser or leave it free

ex : ISBN  : 9782702141557 ou 9782081383784

2 - a problem with accented characters
[bilan-OpenLibrary.txt](https://github.com/textbrowser/biblioteq/files/7728708/bilan-OpenLibrary.txt)

a plus for Open Library: it puts subtitles in addition

**Comment by textbrowser** [comment]
Thanks! Lots of work.

How should the field be ignored? After the parsing, completed fields are highlighted. The person can clear values which are not applicable.

In a future life, I would love if I could instruct the program to produce floating widgets near fields which are questionable and have the person accept those fields if they wish. :)

**Reply by textbrowser** [reply]
Like interactive tool tips.

**Reply by meteos77** [reply]
I take :-)

I explain more in details: no the problem is: if I get after a SRU request almost all the information about the document but it misses for example the subtitle (whereas OpenLibrary gives it to me) but that in the meantime never entered the dimension manually, the openlibrary request overwrites MY information about the dimension whereas it should add the missing information without overwriting the existing information.

in common language : 
in the config : overwrite the fields by OpenLibrary request (YES/NO)

if the title field is empty you fill in the information you have 

if the title field is not empty you overwrite if the user's config asks you = YES

if the title field is not empty you don't overwrite anything if the user's config asks you to = NO

in conclusion if the user wants an update, he deletes himself the field and makes the request (allows to make a copy/paste); if the result is better, he saves it, if it is not what he wants ... he makes a paste

**Reply by textbrowser** [reply]
Those are too many prompts. Something better is required.

I imagine something else. Everything is overridden but with floating tool tips near the fields where the person can revert to the previous value.

For example,

[title] [[x][revert]]

That's a crude representation of the floating tool tip. If I click on the [x], the title is reverted. It's a floating tool tip which is physically attached to its parent widget; moving the window or scrolling the view changes the locations of the tool tips. It's far out and difficult because nothing offers this; but it would be splendid and intuitive with zero configuration. Well, except for maybe deciding which fields to attach the tool tips to.
